test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -75,14 +77,17 @@ class test_ucp_peer_failure : public ucp_test {
 
     void cleanup_rndv_descs();
 
-    void                    *m_sreq, *m_rreq;
-    std::queue<void *>      m_am_rndv_descs;
-    size_t                  m_am_rx_count;
-    size_t                  m_err_count;
-    ucs_status_t            m_err_status;
-    std::string             m_sbuf, m_rbuf;
-    mem_handle_t            m_stable_memh, m_failing_memh;
-    ucs::handle<ucp_rkey_h> m_stable_rkey, m_failing_rkey;
+    void init_buffers(size_t msg_size);
+
+    void                        *m_sreq, *m_rreq;
+    std::queue<void *>          m_am_rndv_descs;
+    size_t                      m_am_rx_count;
+    size_t                      m_err_count;
+    ucs_memory_type_t           m_mem_type;
+    ucs_status_t                m_err_status;
+    std::unique_ptr<mem_buffer> m_sbuf, m_rbuf;
```

Comment:
Why needed to store `unique_ptr` to `mem_buffer` as I can see buffer is structure that already contains pointer to allocated memory. If that is required for `reset()` functionality, I think it is better to realize `mem_buffer::reset()` method for that purpose.

Reply:
imo it is more convinient